lib/ja_serializer/dsl.ex

```diff
@@ -0,0 +1,424 @@
+  on the serializer with a default behavior of `Map.get(struct, :body)`.
+
+      defmodule PostSerializer do
+        use JaSerializer, dsl: true
+        attributes [:title, :body, :html]
+
+        def html(post, _conn) do
+          Earmark.to_html(post.body)
+        end
+      end
+
+  ## Conditional attribute inclusion
+
+  JaSerializer supports the `fields` option as per the JSONAPI spec. This
+  option allows clients to request only the fields they want. For example if
+  you only wanted the html and the title for the post:
+
+      field_param = %{"post" => "title,html", "comment" => "html"}
+
+      # Direct Serialization
+      PostSerializer.format(post, conn, fields: field_param)
+
+      # via PhoenixView integrations from controller
+      render(conn, :show, data: post, opts: [fields: field_param])
+
+  ## Further customization
+
+  Further cusomization of the attributes returned can be handled by overriding
+  the `attributes/2` callback. This can be done in conjuction with the DSL
+  using super, or without the DSL just returning a map.
+
+  """
+  defmacro attributes(atts) do
+    quote bind_quoted: [atts: atts] do
+      # Save attributes
+      @attributes @attributes ++ atts
+
+      # Define default attribute function, make overridable
+      for att <- atts do
+        def unquote(att)(m),    do: Map.get(m, unquote(att))
+        def unquote(att)(m, c), do: apply(__MODULE__, unquote(att), [m])
+        defoverridable [{att, 2}, {att, 1}]
+      end
+    end
+  end
+
+  @doc """
+  Add a has_many relationship to be serialized.
+
+  JSONAPI.org supports three types or relationships:
+
+    * As links - Great for clients lazy loading relationships with lots of data.
+    * As "Resource Indetifiers" - A type/id pair, usefull to relate to data the client already has.
+    * As Included Resources - The full resource is serialized in the same request (also includes Resource Identifiers).
+
+  Links can be combined with either resource identifiers or fully included resources.
+
+  See http://jsonapi.org/format/#document-resource-object-relationships for more
+  details on the spec.
+
+  ## Link based relationships
+
+  Specify a URI or path which responds with the related resource. For example:
+
+      defmodule MyApp.PostView do
+        use JaSerializer
+
+        has_many :comments, link: :comments_link
+        has_one  :author, link: "/api/posts/:id/author"
+
+        def comments_link(post, conn) do
+          MyApp.Router.Helpers.post_comment_url(conn, :index, post.id)
+        end
+      end
+
+  Links can be defined with an atom or string.
+
+  String may be either a relative or absolute path. Path segments beginning
+  with a colon are called as functions on the serializer with the struct and
+  conn passed in. In the above example id/2 would be called which is defined as
+  a default callback.
+
+  When an atom is passed in it is assumed it is a function that will return
```

Comment:
If you decide to change the location atom doc above, you'd want to update this too.